spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/core/AbstractMqttClientManager.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright 2022-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.mqtt.core;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.context.ApplicationEventPublisherAware;
+import org.springframework.context.SmartLifecycle;
+import org.springframework.integration.mqtt.inbound.AbstractMqttMessageDrivenChannelAdapter;
+import org.springframework.util.Assert;
+
+/**
+ * @param <T> MQTT client type
+ * @param <C> MQTT connection options type (v5 or v3)
+ *
+ * @author Artem Vozhdayenko
+ *
+ * @since 6.0
+ */
+public abstract class AbstractMqttClientManager<T, C> implements ClientManager<T, C>, ApplicationEventPublisherAware {
+
+	protected final Log logger = LogFactory.getLog(this.getClass()); // NOSONAR
+
+	private final Set<ConnectCallback> connectCallbacks;
+
+	private final String clientId;
+
+	private boolean manualAcks;
+
+	private ApplicationEventPublisher applicationEventPublisher;
+
+	private String url;
+
+	private volatile T client;
+
+	private String beanName;
+
+	AbstractMqttClientManager(String clientId) {
+		Assert.notNull(clientId, "'clientId' is required");
+		this.clientId = clientId;
+		this.connectCallbacks = Collections.synchronizedSet(new HashSet<>());
+	}
+
+	protected void setManualAcks(boolean manualAcks) {
+		this.manualAcks = manualAcks;
+	}
+
+	protected String getUrl() {
+		return this.url;
+	}
+
+	protected void setUrl(String url) {
+		this.url = url;
+	}
+
+	protected String getClientId() {
+		return this.clientId;
+	}
+
+	protected ApplicationEventPublisher getApplicationEventPublisher() {
+		return this.applicationEventPublisher;
+	}
+
+	protected synchronized void setClient(T client) {
+		this.client = client;
+	}
+
+	protected Set<ConnectCallback> getCallbacks() {
+		return Collections.unmodifiableSet(this.connectCallbacks);
```

Comment:
The method is `protected`, so unlikely going to be used for some outside harm.
Therefore I don't see a reason in extra wrapping to prevent modification.

Reply:
Just the defense if someone decides to implement own client manager incorrectly, but I will remove it, no problem